tests/test_topline_summary.py

```diff
@@ -0,0 +1,309 @@
+from click.testing import CliRunner
+from pyspark.sql import functions as F, Row
+from pyspark.sql.types import *
+
+from mozetl.topline import topline_summary as topline
+
+
+def generate_dates(submission_date_s3, ts_offset=0, creation_offset=0):
+    # convert submission_date into profile_creation_date and timestamps
+    submission = arrow.get(submission_date_s3, "YYYYMMDD")
+    creation = submission.replace(days=+creation_offset)
+    timestamp = submission.replace(days=+ts_offset)
+
+    # variables for conversion
+    epoch = arrow.get(0)
+    seconds_in_day = 60 * 60 * 24
+    nanoseconds_in_second = 10**9
+
+    date_snippet = {
+        "submission_date_s3": submission_date_s3,
+        "profile_creation_date": (
+            long((creation - epoch).total_seconds() / seconds_in_day)
+        ),
+        "timestamp": (
+            long((timestamp - epoch).total_seconds() * nanoseconds_in_second)
+        )
+    }
+
+    return date_snippet
+
+
+def search_row(engine='hooli', count=1, source='searchbar'):
+    return Row(
+        engine=unicode(engine),
+        source=unicode(source),
+        count=count
+    )
+
+
+schema = StructType([
+    StructField("client_id", StringType(), True),
+    StructField("timestamp", StringType(), True),
+    StructField("is_default_browser", BooleanType(), True),
+    StructField("search_counts",
+                ArrayType(
+                    StructType([
+                        StructField("engine", StringType(), True),
+                        StructField("source", StringType(), True),
+                        StructField("count", LongType(), True)
+                    ]), True),
+                True),
+    StructField("country", StringType(), True),
+    StructField("profile_creation_date", LongType(), True),
+    StructField("normalized_channel", StringType(), True),
+    StructField("os", StringType(), True),
+    StructField("subsession_length", LongType(), True),
+    StructField("submission_date_s3", StringType(), True),
+])
+
+
+start_ds = "20170601"
+end_ds = "20170608"
+
+default_sample = {
+    "client_id": "client-id",
+    "timestamp": long(1.4962752e+18),
+    "is_default_browser": True,
+    "search_counts": [search_row()],
+    "country": "US",
+    "profile_creation_date": 17318,
+    "normalized_channel": "release",
+    "os": "Linux",
+    "subsession_length": 3600,
+    "submission_date_s3": "20170601",
+}
+default_sample.update(generate_dates(start_ds))
+
+
+@pytest.fixture()
+def generate_data(dataframe_factory):
+    return functools.partial(
+        dataframe_factory.create_dataframe,
+        base=default_sample,
+        schema=schema
+    )
+
+
+@pytest.fixture()
+def cool_df(dataframe_factory):
+    rows = [
+        # exact
+        "ice",
+        "lice",
+        # prefixed
+        "glacier water",
+        "melting glaciers",
+        # suffixed
+        "gentoo penguin",
+        "penguin suit",
+        # anywhere
+        "dinosaurs are always cool",
```

Comment:
Please add another example row that has `dinosaurs` somewhere in the middle to differentiate from the prefix/suffix case.